pyblazing/pyblazing/apiv2/context.py

```diff
@@ -226,49 +226,36 @@ def collectPartitionsPerformPartition(
         nodes,
         ctxToken,
         input,
-        dask_mapping,
+        partition_keys_mapping,
         df_schema,
         by,
-        i):  # this is a dummy variable to make every submit unique which is necessary
+        i):  
     import dask.distributed
-    worker_id = dask.distributed.get_worker().name
-    if(isinstance(input, dask_cudf.core.DataFrame)):
-        partitions = dask_mapping[worker_id]
-        if (len(partitions) == 0):
-            input = df_schema
-        elif (len(partitions) == 1):
-            input = input.get_partition(partitions[0]).compute()
+    worker = dask.distributed.get_worker()
+    worker_id = nodes[i]['worker']
+
+    if worker_id in partition_keys_mapping:
+        partition_keys = partition_keys_mapping[worker_id]
+        if len(partition_keys) > 1:
+            node_inputs = []
+            for key in partition_keys:
+                node_inputs.append(worker.data[key])
+            # TODO, eventually we want the engine side of the partition function to handle the table in parts
+            node_input = cudf.concat(node_inputs) 
```

Comment:
This isn't necessary you can just have the table scan machine be able to receive multiple node_inputs

Reply:
k